File: 150/main.go
package main

import (
	"strconv"
)

func evalRPN(tokens []string) int {
	stack := []int{}

	for _, v := range tokens {
		switch v {
		case "+":
			a, b := stack[len(stack)-1], stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, a+b)
		case "-":
			a, b := stack[len(stack)-1], stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, b-a)
		case "*":
			a, b := stack[len(stack)-1], stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, a*b)
		case "/":
			a, b := stack[len(stack)-1], stack[len(stack)-2]
			stack = stack[:len(stack)-2]
			stack = append(stack, b/a)
		default:
			num, _ := strconv.Atoi(v)
			stack = append(stack, num)
		}
	}

	return stack[0]
}

func main() {
	println(evalRPN([]string{"2", "1", "+", "3", "*"}))
	println(evalRPN([]string{"4", "13", "5", "/", "+"}))
}

File: 150/main.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        stack = []
        for v in tokens:
            match v:
                case "+":
                    stack.append(int(stack.pop()) + int(stack.pop()))
                case "-":
                    a, b = int(stack.pop()), int(stack.pop())
                    stack.append(b - a)
                case "*":
                    stack.append(int(stack.pop()) * int(stack.pop()))
                case "/":
                    a, b = int(stack.pop()), int(stack.pop())
                    stack.append(int(b / a))
                case _:
                    stack.append(int(v))

        return int(stack[0])


s = Solution()
print(s.evalRPN(["2", "1", "+", "3", "*"]))
print(s.evalRPN(["4", "13", "5", "/", "+"]))
